modules/core/src/smithy4s/Document.scala

```diff
@@ -105,9 +105,21 @@ object Document {
     def fromSchema[A](
         schema: Schema[A],
         cache: Cache
+    ): Encoder[A] =
+      fromSchema(schema, cache, explicitDefaultsEncoding = false)
+
+    def fromSchema[A](
+        schema: Schema[A],
+        cache: Cache,
+        explicitDefaultsEncoding: Boolean
```

Comment:
Similarly to my comment [here](https://github.com/disneystreaming/smithy4s/pull/1315/files#r1442591876), this method should be removed in favour of a `withExplicitDefaultsEncoding(explicitDefaults: Boolean)` on the `Encoder`. 

This implies the creation of public interface. You can take [this](https://github.com/disneystreaming/smithy4s/blob/cebdd437b02608b9f9e834c9b1b03c0a37036d5a/modules/json/src/smithy4s/json/JsoniterCodecCompiler.scala#L31) as inspiration.

Reply:
It's not a bug, it's a feature 😄 

If you want to compile a codec without cache, you should be able to call 

```
Json.jsoniter.withMaxArity(1).fromSchema(Schema.string)
```

Otherwise, if you wanted to use a shared cache across several codec compilations, you would have to do : 

```
val compiler = Json.jsoniter.withMaxArity(1) 
val cache = compiler.createCache() 
val stringCodec = compiler.fromSchema(Schema.string, cache) 
```